packages/plugin-vue/src/main.ts

```diff
@@ -212,6 +213,13 @@ export async function transformMain(
     code: resolvedCode,
     map: resolvedMap || {
       mappings: ''
+    },
+    meta: {
+      vite: {
+        lang: isTs
+          ? 'ts'
+          : descriptor.scriptSetup?.lang || descriptor.script?.lang || 'js'
+      }
```

Comment:
Check `isTs` first as there's a chance `script lang` and `script setup lang` are using different lang.

I also evaluated `descriptor.scriptSetup?.lang` first, then `descriptor.script?.lang` as it's more likely that one sets `script setup`?

Happy to refactor this part if there's more that needs to change.

Reply:
I think we could simply do `isTS ? 'ts' : 'js'` as [`<script setup>` and `<script>` have to have same language type](https://github.com/vuejs/core/blob/4e7b36f5c3ee72cb9554982dd174f1989d5488f6/packages/compiler-sfc/src/compileScript.ts#L243).

Reply:
Ah nice find. I think I could do `descriptor.script?.lang || descriptor.scriptSetup?.lang || 'js'` directly then. `isTs` might not be needed here, and the Vue compiler seems to favor checking `script` first, then `scriptSetup`

Reply:
Now simplified to

```js
meta: {
  vite: {
    lang: descriptor.script?.lang || descriptor.scriptSetup?.lang || 'js'
  }
}
```